runelite-mixins/src/main/java/net/runelite/mixins/RSClientMixin.java

```diff
@@ -483,6 +485,20 @@ public static void gameStateChanged(int idx)
 		eventBus.post(gameStateChange);
 	}
 
+	@Inject
+	@FieldHook("grandExchangeOffers")
+	public static void onGrandExchangeOffersChanged(int idx)
+	{
+		// Not sure if I need a check for idx=-1 here... -- InternetPerson
```

Comment:
Looking at the current client, this does not happen as the field is only initialized on clinit (which will not trigger an event). Maybe it wouldn't hurt to add a check for this anyway.